kafka-connect-s3/src/main/java/io/confluent/connect/s3/storage/S3ParquetOutputStream.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2017 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.connect.s3.storage;
+
+import com.amazonaws.services.s3.AmazonS3;
+import io.confluent.connect.s3.S3SinkConnectorConfig;
+
+import java.io.IOException;
+
+public class S3ParquetOutputStream extends S3OutputStream {
+
+  private volatile boolean commit;
+
+  public S3ParquetOutputStream(String key, S3SinkConnectorConfig conf, AmazonS3 s3) {
+    super(key, conf, s3);
+    commit = false;
+  }
+
+  @Override
+  public void close() throws IOException {
+    if (commit) {
+      super.commit();
+    } else {
```

Comment:
yes, that's what I had in mind. Let's unset the `commit` flag right after we commit. This should run in the same thread, so no race condition w.r.t. resetting this flag.